src/server/node_manager.c

```diff
@@ -6839,6 +6839,51 @@ set_nodes(void *pobj, int objtype, char *execvnod_in, char **execvnod_out, char
 	return 0;
 }
 
+void
+disassociate_job_from_moms(job *pjob, struct pbsnode *pnode)
```

Comment:
I don't think we need this, we can put this code in `deallocate_job_from_node()`

Reply:
Thanks, @bremanandjk for the explanation. I got it that we can't put this in `deallocate_job_from_node()`. But I would suggest to make it static and name it like `remove_job_index_from_mom()` (as we have `add_job_index_in_mom()` in the same file, which is exactly reverse of this). Also please add function header.

Reply:
I have renamed the function name.